### PR TITLE
treefmt: init at 0.1.1

### PR DESCRIPTION
**Pull request by zimbatm**
###### Motivation for this change

In the long term, format all of nixpkgs.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by zimbatm** [comment]
Result of `nixpkgs-review pr 120521` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>treefmt</li>
  </ul>
</details>

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -30915,6 +30915,8 @@ in
 
   fac-build = callPackage ../development/tools/build-managers/fac {};
 
+  treefmt = callPackage ../development/tools/treefmt {};
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:30918]:
```suggestion
  treefmt = callPackage ../development/tools/treefmt { };
```

File: pkgs/development/tools/treefmt/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  };
+
+  cargoSha256 = "08k60gd23yanfraxpbw9hi7jbqgsxz9mv1ci6q9piis5742zlj9s";
+
+  meta = {
+    description = "one CLI to format the code tree";
+    homepage = "https://github.com/numtide/treefmt";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ zimbatm ];
+    platforms = lib.platforms.unix;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/treefmt/default.nix:20]:
```suggestion
```

set by buildRustPackage to the right subset of unix.

File: pkgs/development/tools/treefmt/default.nix [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{ lib, rustPlatform, fetchFromGitHub }:
+rustPlatform.buildRustPackage rec {
```
Comment (SuperSandro2000) [inline pkgs/development/tools/treefmt/default.nix:2]:
```suggestion
{ lib, rustPlatform, fetchFromGitHub }:

rustPlatform.buildRustPackage rec {
```

Comment (zimbatm) [reply pkgs/development/tools/treefmt/default.nix:2]: I don't see the purpose of this one. What advantage is there to have an additional empty line?

**Comment by zimbatm** [comment]
Thanks for your review @SuperSandro2000, I'll go ahead and merge it like this since it's functional.

File: pkgs/development/tools/treefmt/default.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/development/tools/treefmt/default.nix:2]: It clearly sepearates inputs from the actual package.

Comment (zimbatm) [reply pkgs/development/tools/treefmt/default.nix:2]: Syntax highlighting already does this for me. I think this is more a matter of habit?

**Pull request merged**